### PR TITLE
Add simple test for container + fix basics so that Jenkins can start

### PR DESCRIPTION
**Pull request by batmat**
Jenkins should be available after startup.

I have the changes locally that makes this work, though not clean (at all) yet:

```
evergreen $ ./test-container.sh
Using the port 5901
Start container under test and wait a bit for its startup:
b194fb14217a1174a43d888d70942328161419e21fdaaeee2688a9fc967a05d7
Connect to Jenkins
Check content
```

cc @rtyler

### CODE REVIEW

**Comment by rtyler** [comment]
@batmat won't this fail every pull request until the evergreen-client is properly downloading the `jenkins.war` and starting it?

**Comment by batmat** [comment]
@rtyler yes, I meant it as a first commit. It is expected to fail as is indeed. Going to see how to push a bare minimum to make it pass (probably with some `FIXME` hacks)

**Comment by rtyler** [comment]
Heh, the Pipeline didn't fail, so perhaps the test is bad? :smile:

**Comment by batmat** [comment]
THAT WAS THE INTENT, AND EXPECTED :+1: 

/me goes back fixing this crap. ;-)

**Comment by batmat** [comment]
Better, **now** it fails as expected.
image

File: supervisord.conf [hunk 1]
```diff
@@ -8,7 +8,7 @@ port=:9001
 supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface
 
 [program:jenkins]
-command=/usr/local/bin/jenkins.sh
+command=/usr/local/bin/shim-startup-wrapper.sh
```
Comment (batmat) [inline supervisord.conf]: @rtyler WDYT of this as a temporary measure to be able to move forward? 

At least it's very useful for me locally to be able to iterate.

File: Dockerfile [hunk 2]
```diff
@@ -53,6 +60,9 @@ RUN mkdir -p /evergreen
 COPY client /evergreen/client
 COPY essentials.yaml /evergreen
 
+# FIXME REMOVE (to ease iteration/speed just for now), see also shim-startup-wrapper.sh
+RUN wget --quiet http://mirrors.jenkins.io/war-stable/latest/jenkins.war -O /usr/share/jenkins/jenkins.war
```
Comment (batmat) [inline Dockerfile]: complementary to the shim. The goal here is to try and leverage Docker cache and avoid having to download the WAR **each** time one would start the container. Which is already a few seconds on a high bandwidth network, but can kill everything on a more flaky network when hacking on evergreen.

**Pull request merged**